characters/playermelon.py

```diff
@@ -0,0 +1,235 @@
+                state.screen_shaking = True
+
+        self.animate_flame()
+
+        if self.velocity > 0:
+            audio.stop_infinite_sfx('beepbeepbeep')
+            audio.play_infinite_sfx('rocket', volume=(abs(self.velocity) / self.max_velocity))
+        elif self.velocity < 0:
+            audio.stop_infinite_sfx('rocket')
+            audio.play_infinite_sfx('beepbeepbeep', volume=(abs(self.velocity) / self.max_velocity))
+        else:
+            audio.stop_infinite_sfx('rocket')
+            audio.stop_infinite_sfx('beepbeepbeep')
+
+        if (self.health / self.max_health) < .25:
+            audio.play_infinite_sfx('heartbeat')
+        else:
+            audio.stop_infinite_sfx('heartbeat')
+
+    def animate_flame(self):
+        velocity_based_animation_rate = 2 - int(abs(self.velocity) / 4)
+        self.flame_frame_timer += 1
+        if self.flame_frame_timer > velocity_based_animation_rate:
+            self.flame_frame_timer = 0
+            self.flame_frame += 1
+            if self.flame_frame > 3:
+                self.flame_frame = 0
+
+    def draw_flame(self, screen):
+        image_height = int(abs(self.velocity)) + 10
+        flame_subsurface = self.flame_image.subsurface(self.flame_frame * 20, 20 - image_height, 20, image_height)
+        rotated_image = pygame.transform.rotate(flame_subsurface, self.angle - 180)
+        w, h = rotated_image.get_size()
+
+        distance_from_center = (20 + int(abs(self.velocity) / 2))
+        camera.blit(rotated_image, (self.x - distance_from_center * self.orientation_vector_x - w / 2,
+                                    self.y - distance_from_center * self.orientation_vector_y - h / 2))
+
+    def draw_hud(self, screen):
+        # health bar
+        pygame.draw.rect(screen, (255, 0, 0), (10, 10, 100, 10))
+        if self.health > 0:
+            pygame.draw.rect(screen, (0, 255, 0), (10, 10, 100 * self.health / self.max_health, 10))
+
+        # velocity bar
+        pygame.draw.rect(screen, (255, 0, 0), (10, 370, 10, 100))
+        bar_height = 100 * abs(self.velocity) / self.max_velocity
+        if bar_height > 0:
+            color = (0, 255, 0) if self.velocity > 0 else (255, 192, 0)
+            pygame.draw.rect(screen, color, (10, 470 - bar_height, 10, bar_height))
+
+    # Draw the watermelon and heath bar and potentialy game over
+    def draw(self, screen):
+        super().draw(screen)
+        self.draw_melon(screen)
+        self.draw_flame(screen)
+
+        # TODO: Move these displays to a HUD class
+        self.draw_hud(screen)
+
+        if self.health <= 0:
+            state.fade_to('game_over')
+
+    def create_collider(self):
+        return Collider(self, effect=CollisionEffect.Player, reaction=self.collided)
+
+    def collided(self, collider, effect):
+        super().collided(collider, effect)
+
+        if effect == CollisionEffect.Halt:
+            self.acceleration = self.move_velocity = self.velocity = self.health = 0
```

Comment:
What's the point of health if a single drumstick kills you?

Reply:
:man_shrugging:

That's how it was before, I think it's mostly about the health drain when you shoot seeds...